File: app/src/main/java/com/kota/Telnet/TelnetSocketChannel.java
package com.kota.Telnet;

import java.io.IOException;
import java.nio.ByteBuffer;

public interface TelnetSocketChannel {
    boolean finishConnect() throws IOException;

    int read(ByteBuffer byteBuffer) throws IOException;

    int write(ByteBuffer byteBuffer) throws IOException;
}

File: app/src/main/java/com/kota/Telnet/TelnetSocketChannel.kt
package com.kota.Telnet

import java.io.IOException
import java.nio.ByteBuffer

interface TelnetSocketChannel {
    @Throws(IOException::class)
    fun finishConnect(): Boolean

    @Throws(IOException::class)
    fun read(byteBuffer: ByteBuffer?): Int

    @Throws(IOException::class)
    fun write(byteBuffer: ByteBuffer?): Int
}
